blockchain/types/account/account_serializer.go

```diff
@@ -64,14 +65,7 @@ func (ser *AccountSerializer) GetAccount() Account {
 
 func (ser *AccountSerializer) DecodeRLP(s *rlp.Stream) error {
 	if err := s.Decode(&ser.accType); err != nil {
-		// fallback to decoding a LegacyAccount object.
-		acc := newLegacyAccount()
-		if err := s.Decode(acc); err != nil {
-			return err
-		}
-		ser.accType = LegacyAccountType
-		ser.account = acc
-		return nil
+		return err
```

Comment:
Instead of modifying this code which is used by other many code, we need to consider a storage trie iterator for this API like below. @ehnuje 

```
		dataTrie, err := it.state.db.OpenStorageTrie(pa.GetStorageRoot())
		if err != nil {
			return err
		}
		it.dataIt = dataTrie.NodeIterator(nil)
		if !it.dataIt.Next(true) {
			it.dataIt = nil
		}
```